### PR TITLE
chore: make open api lint check as optional on choosen file

### PR DESCRIPTION
**Pull request by heliprokupetz14**

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -2,6 +2,12 @@ name: pull_request
 
 on: 
   workflow_call:
+    inputs:
+      openApi:
+        description: 'Path to the OpenAPI file'
+        type: string
+        required: true
```
Comment (AnguIar) [inline .github/workflows/pull_request.yaml:9]: it's not a required value, change to `false`

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -2,6 +2,12 @@ name: pull_request
 
 on: 
   workflow_call:
+    inputs:
+      openApi:
```
Comment (AnguIar) [inline .github/workflows/pull_request.yaml:6]: Rename to `openApiFilePath`

File: .github/workflows/pull_request.yaml [hunk 3]
```diff
@@ -30,24 +36,26 @@ jobs:
           eslint_extensions: ts
 
       - name: OpenAPI Lint Checks
+        if: ${{ github.event.inputs.openApi.required == 'true' }}
```
Comment (AnguIar) [inline .github/workflows/pull_request.yaml:39]: This condition is wrong. You should add another argument (other than `openApiFilePath`) and execute this step only if this variable's value is `true`.

File: .github/workflows/pull_request.yaml [hunk 4]
```diff
@@ -30,24 +36,26 @@ jobs:
...
+      - name: Run Snyk to check for vulnerabilities
+        uses: snyk/actions/node@master
+        continue-on-error: true
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
 
   tests:
     name: Run Tests
     runs-on: ubuntu-latest
+    needs: eslint
```
Comment (AnguIar) [inline .github/workflows/pull_request.yaml:58]: Why did you add it?

**Review by AnguIar — CHANGES_REQUESTED** [review]
Good job!:)

File: .github/workflows/pull_request.yaml (hunk 3 above)
Comment (heliprokupetz14) [reply .github/workflows/pull_request.yaml:39]: Did you mean to have an argument of the openApiFilePath and an argument of the openApi file name?

**Review by netanelC — CHANGES_REQUESTED** [review]
There are unnecessary whitepsaces

**Review by netanelC — APPROVED** [review]
Good Job

**Review by AnguIar — APPROVED** [review]
